### ISSUE TITLE
google_compute_network.auto_create_subnetworks field forces creation of new resource

### ISSUE THREAD

**Issue by julianvmodesto** [terraform-providers/terraform-provider-google#1752]
### Terraform Version
```
$ terraform -v
Terraform v0.11.7
+ provider.google v1.15.0
+ provider.random v1.3.1
```
### Affected Resource(s)
- google_compute_network

### Terraform Configuration Files
```hcl
resource "google_compute_network" "self" {
  name        = "${var.network_name}"
  description = "Default network for the project"

  project = "${google_project_service.self_compute.project}"

  auto_create_subnetworks = false
  routing_mode            = "REGIONAL"
}
```

### Debug Output
n/a

### Panic Output
n/a

### Expected Behavior
Changing the `auto_create_subnetworks` field should not force creation of a new resource.

### Actual Behavior
Changing the `auto_create_subnetworks` field forces creation of a new resource.

Also, I'm not sure if this is an issue, but the inaccessible `id` field is computed but also forces a new resource.

See the plan output below.

```diff
-/+ module.my_project.google_compute_network.self (new resource required)
      id:                        "default" => <computed> (forces new resource)
      auto_create_subnetworks:   "true" => "false" (forces new resource)
      description:               "Default network for the project" => "Default network for the project"
      gateway_ipv4:              "" => <computed>
      name:                      "default" => "default"
      project:                   "my-project" => "my-project"
      routing_mode:              "REGIONAL" => "REGIONAL"
      self_link:                 "https://www.googleapis.com/compute/v1/projects/my-project/global/networks/default" => <computed>
```

### Steps to Reproduce
Flip the `auto_create_subnetworks` field.

### Important Factoids
n/a

### References
n/a

**Comment by paddycarver** [comment]
According to the docs, `auto_create_subnetworks` only matters for newly created networks:

> In "auto subnet mode", a **newly created network** is assigned the default CIDR of 10.128.0.0/9 and it automatically creates one subnetwork per region.

(emphasis mine)

So I'm not sure what updating this in-place would do, or if that's even possible. Is this something you can do with another tool/the console?

**Comment by julianvmodesto** [comment]
Yes, I've manually switched from `default` to `custom` auto subnet creation mode in the console.

Switching from `default` to `custom` auto subnet mode is the only in-place switch that is available -- the `custom` to `default` switch isn't available.

This is the description of the change in a confirmation prompt that I saw:

```
Switching VPC network mode

Switching from auto-subnet to custom-subnet VPC network allows you to manually define subnet prefixes for each region in your VPC network. You cannot undo this action.

Are you sure you want to change VPC network 'default' to custom subnet network?
```

Is this switch available via the API?

**Comment by tmshn** [comment]
ref: https://cloud.google.com/compute/docs/reference/rest/v1/networks/switchToCustomMode

**Comment by AlfatahB** [comment]
b/262334608

**Comment by AlfatahB** [comment]
I've looked into this issue and the following are some of the details that I've found out:
- It occurs due to the `ForceNew` [behavior](https://github.com/hashicorp/terraform-provider-google/blob/main/google/resource_compute_network.go#L58) of the `auto_create_subnetworks` field in the `google_compute_network` resource.
- This issue seems to add support for the in-place updation of the `auto_create_subnetworks` field.
- There is a [separate endpoint](https://cloud.google.com/compute/docs/reference/rest/v1/networks/switchToCustomMode) to switch the network mode from auto subnet mode to custom subnet mode. But, there is no endpoint to switch the network mode from custom subnet mode to auto subnet mode. Also, we generally prefer for APIs to just support updating with the `patch` or `update` methods because having a field-specific method to call is harder for us to maintain.

A possible solution can be to make the `auto_create_subnetworks` field to use this [separate endpoint](https://cloud.google.com/compute/docs/reference/rest/v1/networks/switchToCustomMode) to switch the network mode from auto subnet mode to custom subnet mode and we can make terraform throw an error if we switch from custom subnet mode to auto subnet mode. Alternatively, we can also make terraform to trigger `diff.ForceNew("auto_create_subnetworks")` if we switch from custom subnet mode to auto subnet mode.

**Comment by rileykarson** [comment]
As mentioned above, this makes sense to allow transitioning an auto subnet to a custom one, but we'd want to continue to recreate in the other direction with `diff.ForceNew("auto_create_subnetworks")`.

**Comment by jbbqqf** [comment]
Hi! Triaging older `google_compute_network` reports — the ForceNew behavior on `auto_create_subnetworks` reflects an API constraint, not a provider-side limitation.

The Compute Networks API does not accept changes to `autoCreateSubnetworks` on an existing network — there's no PATCH path that converts a legacy/auto-mode network into custom-mode (or vice-versa) in place. The provider mirrors that with `immutable: true` in the mmv1 source ([`mmv1/products/compute/Network.yaml#L149`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/compute/Network.yaml#L149)), which generates the `ForceNew` you're seeing.

Google does have a separate documented procedure to convert auto-mode → custom-mode (see [Convert a network from auto mode to custom mode](https://cloud.google.com/vpc/docs/create-modify-vpc-networks#convert-auto-to-custom)), but it's an operations-level workflow, not a single API call — so it can't be wired as a Terraform in-place update.

API docs: https://cloud.google.com/compute/docs/reference/rest/v1/networks/patch

Would you mind closing? If GCP exposes a one-shot PATCH for this in the future, that's the time to revisit.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the mmv1 immutability flag and the upstream API behavior were verified manually.)
